### ISSUE TITLE
RRHO2 hyper overlap produces NaN while RRHO does not

### ISSUE THREAD

**Issue by gdevenyi** [RRHO2/RRHO2#9]
Running RRHO2_initalize on my dataset such as this:
```R
> RRHO2_results = RRHO2_initialize(human_rel_rrho, chimp_rel_rrho,  stepsize=100,  multipleTesting="BH",  labels = c("human","chimp"))
There were 50 or more warnings (use warnings() to see the first 50)
> warnings()
Warning messages:
1: In phyper(q = count - 1, m = a, n = n - a + 1, k = b,  ... : NaNs produced
2: In phyper(q = count - 1, m = a, n = n - a + 1, k = b,  ... : NaNs produced
<clip>
```

RRHO package with similar invocataion doesn't have such issues
```
RRHO_results = RRHO(human_rel_rrho, chimp_rel_rrho,  stepsize=25, alternative = "two.sided", BY=TRUE, outputdir=getwd(), plots=TRUE, labels = c("human","chimp"))
```

Data is a column with gene names and a column with -log(p-value)*sign(effect):
```R
> summary(chimp_rel_rrho)
      chimp_rel[, 1]      value           
 A_23_P100595:    1   Min.   :-276.04974  
 A_23_P10091 :    1   1st Qu.:  -4.53327  
 A_23_P102331:    1   Median :   0.02146  
 A_23_P102816:    1   Mean   :  -0.13997  
 A_23_P103269:    1   3rd Qu.:   4.72286  
 A_23_P103405:    1   Max.   : 247.64899  
 (Other)     :19094    
> summary(human_rel_rrho)
      human_rel[, 1]      value           
 A_23_P100595:    1   Min.   :-17.161466  
 A_23_P10091 :    1   1st Qu.: -0.623182  
 A_23_P102331:    1   Median :  0.000023  
 A_23_P102816:    1   Mean   : -0.125274  
 A_23_P103269:    1   3rd Qu.:  0.549578  
 A_23_P103405:    1   Max.   : 15.345104  
 (Other)     :19079         
```

**Comment by Caleb-Huo** [comment]
Thanks for reporting this issue.
Is it possible for me to have your data? Or a minimum data that I can reproduce this error?
The example data has similar structure, but it works.

**Comment by gdevenyi** [comment]
Happy to provide privately, you can email me gdevenyi@gmail.com

**Comment by gdevenyi** [comment]
See your email on your github profile page. Thanks for your help.

**Comment by Caleb-Huo** [comment]
Thanks for sharing your example data.
I fixed three issues:
1, NA values should not be used as input. A validation procedure was added to forbid NA value.
2, The gene names should be same (not necessary to be in the same order) between the two input lists. A validation procedure was added to check for this.
3, If the overlapping p-values are extreme (i.e., p<1e-1000), multiple testing procedure will generate exact 0. Thus log(adjusted-p) = Inf, which is not a valid input for the heatmap. Thus, a warning message will be generated to suggest to use raw p-value.
Let me know if there are still any other problems.
Thanks again for reporting these issues.
